theano/tensor/extra_ops.py

```diff
@@ -11,6 +11,52 @@
 tensor = basic
 
 
+class CpuContiguous(theano.Op):
+    """
+    Check to see if the input is c-contiguous,
+    if it is, do nothing, else return a contiguous array
+    """
+    __props__ = ()
+    
```

Comment:
This needs a view_map that declares the output is a view on the input `{0: [0]}`